core/src/main/java/org/apache/struts2/interceptor/csp/DefaultCspSettings.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.struts2.interceptor.csp;
+
+import com.opensymphony.xwork2.ActionContext;
+
+import javax.servlet.http.HttpServletResponse;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Map;
+
+
+public class DefaultCspSettings implements CspSettings {
+
+    private static final int NONCE_LENGTH = 18;
+    private final SecureRandom sRand  = new SecureRandom();
+    private String reportUri = "csp-reports";
```

Comment:
this value should probably start with a /, otherwise I suspect the report URI will be relative to where the violation occurs. So if there's a script with no nonce in `/shop/cart`, this might try to send a report to `/shop/csp-reports` instead of `/csp-reports`

Reply:
The one problem with that might be: struts URLs seem to always have the project name - 
Whenever we add an endpoint in the vulnerable application for example we can find it when we go to `http://localhost:8080/maven-example/endpoint` on the browser, with `maven-example` being the project name declared in pom.xml as `<finalName>` 
When I run the application with `/csp-reports` it sends the reports to `http://localhost:8080/csp-reports` which doesn't work because the CspReportAction is on `http://localhost:8080/maven-example/csp-reports`.
Maybe we can add a check in `addCspHeaders()` to add the project name (if there's a way to get the project name) to the `reportUri` if it's set to a relative path? Would there be exceptions to this situation when we wouldn't want this processing to happen? Is there a better solution we can consider?

Reply:
I found this [https://stackoverflow.com/questions/2204870/how-to-get-domain-url-and-application-name](url) but it is a bit complicated to find the right location to enter the path

Reply:
resolved the issue

Reply:
resolved!